scripts/utils/calculate-ARI.R

```diff
@@ -92,3 +92,101 @@ calculate_ari <- function(integrated_sce,
+#'    to create `integrated_sce`. One of: fastMNN, harmony, rpca, cca, scvi, or scanorama
+#' @param unintegrated Indicates whether the provided data is integrated (`FALSE`; default) or
+#'   integrated (`TRUE`).
+#' @param batch_column The variable in `integrated_sce` indicating the grouping of interest.
+#'  Generally this is either batches or cell types. Default is "batch".
+#'
+#' @return Data frame with three columns: `ari`, the calculated ARI, 
+#'   the corresponding `batch_id` found in the `batch_column` 
+#'   for the original SCE object, and the `integration_method`, the provided integration method
+#'
+calculate_reverse_ari <- function(individual_sce_list,
+                                  integrated_sce,
+                                  seed = NULL,
+                                  integration_method = NULL,
+                                  unintegrated = FALSE,
+                                  batch_column = "batch"){
+  set.seed(seed)
+  
+  # check that list of SCE objects is named
+  batch_ids <- names(individual_sce_list)
+  if(is.null(batch_ids)){
+    stop("Must provide a named list of SCE objects.")
+  } else {
+    # make sure the batch ids provided match between the list and the integrated object 
+    if(!all(batch_ids %in% unique(colData(integrated_sce)[, batch_column]))){
+      stop("Names of provided SCE objects included in the individual SCE object 
+           do not match batch IDs present in the batch_column of the integrated object")
+    }
+  }
+  
+  # Settings depending on whether data is integrated or not
+  if (unintegrated){
+    # In the end, we'll return "unintegrated" in the data frame for integration method
+    integration_method <- "unintegrated"
+    
+    # use simply "PCA" for reduced dimensions
+    reduced_dim_name <- "PCA"
+  } else {
+    # Check integration method
+    integration_method <- check_integration_method(integration_method)
+    
+    # Get name for reduced dimensions
+    reduced_dim_name <- get_reduced_dim_name(integration_method)
+  }
+
+  
+  # get pcs for individual objects
+  ind_pcs <- purrr::map(individual_sce_list, get_individual_pcs)
+  
+  # Pull out the PCs or analogous reduction from integrated object
+  integrated_pcs <- reducedDim(integrated_sce, reduced_dim_name)
+  
+  # cluster integrated pcs only one time
+  integrated_clustering_result <- bluster::clusterRows(integrated_pcs, 
+                                                       bluster::NNGraphParam(cluster.fun = "louvain",
+                                                                             type = "jaccard",
+                                                                             k = k)) |> 
+    set_names(rownames(integrated_pcs)) # make sure to set the names with the batch ids
+  
+  
+  # for every batch id, cluster and then calculate ari for that batch 
+  all_ari <- batch_ids |> 
+    purrr::map_chr(\(batch){
```

Comment:
Since ari is numeric, why return as `chr`? Maybe `map_dbl()` here?
```suggestion
    purrr::map_dbl(\(batch){
```